File: frontend/src/types/project.ts
// 短剧项目阶段：用于表示项目当前处在生产链路的哪一步。
export type ShortDramaProjectStage = 'planning' | 'scripting' | 'storyboard' | 'localization' | 'material' | 'launch' | 'completed'

// 短剧项目状态：归档是软删除状态，不会物理删除项目。
export type ShortDramaProjectStatus = 'active' | 'paused' | 'completed' | 'archived'

// 项目优先级：用于后台列表快速判断推进顺序。
export type ShortDramaProjectPriority = 'high' | 'medium' | 'low'

// 后端返回的短剧项目结构，字段与数据库模型保持一致。
export type ShortDramaProject = {
  id: number
  name: string
  genre: string | undefined
  target_market: string | undefined
  language: string
  episode_count: number
  stage: ShortDramaProjectStage
  description?: string | null
  owner?: string | null
  priority: ShortDramaProjectPriority
  status: ShortDramaProjectStatus
  created_at: string
  updated_at: string
}

export type ShortDramaProjectCreate = {
  name: string
  genre: string | undefined
  target_market: string | undefined
  language: string
  episode_count: number
  stage: ShortDramaProjectStage
  description?: string | null
  owner?: string | null
  priority: ShortDramaProjectPriority
  status: ShortDramaProjectStatus
}

export type ShortDramaProjectUpdate = Partial<ShortDramaProjectCreate>

export type ShortDramaProjectListParams = {
  keyword?: string
  genre?: string | undefined
  stage?: ShortDramaProjectStage | undefined
  status?: ShortDramaProjectStatus | undefined
  skip?: number
  limit?: number
}

export type ShortDramaProjectListResponse = {
  items: ShortDramaProject[]
  total: number
}

// 项目详情统计：用于项目工作台展示各生产环节资产数量。
export type ProjectStats = {
  content_plan_count: number
  episode_count: number
  script_count: number
  storyboard_count: number
  localization_count: number
  ad_material_count: number
  media_asset_count: number
}

// 项目生产链路节点：描述每个 AI 短剧生产阶段的状态和资产数量。
export type ProjectPipelineItem = {
  key: string
  title: string
  status: 'completed' | 'processing' | 'pending'
  count: number
  description: string
}

// 最近数据条目：用于展示最近分镜、广告素材和媒体资产。
export type ProjectRecentItem = {
  id: number
  title: string
  type: string
  status: string
  created_at: string
  episode_id?: number | null
  episode_no?: number | null
}

// 项目详情聚合数据：一个接口支撑项目详情页的基础信息、统计、链路和最近数据。
export type ProjectOverview = {
  project: ShortDramaProject
  stats: ProjectStats
  pipeline: ProjectPipelineItem[]
  recent_storyboards: ProjectRecentItem[]
  recent_ad_materials: ProjectRecentItem[]
  recent_media_assets: ProjectRecentItem[]
}

File: backend/app/schemas/project.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ShortDramaProjectBase(BaseModel):
    # 项目基础字段：与数据库模型保持一致，前后端都围绕这些字段做项目管理。
    name: str
    genre: str
    target_market: str
    language: str
    episode_count: int = 60
    stage: str = "planning"
    description: Optional[str] = None
    owner: Optional[str] = None
    priority: str = "medium"
    status: str = "active"


class ShortDramaProjectCreate(ShortDramaProjectBase):
    # 新建项目时允许前端不传阶段、状态和优先级，由后端默认填充。
    pass


class ShortDramaProjectUpdate(BaseModel):
    # 修改项目时所有字段可选，只更新用户传入的字段。
    name: Optional[str] = None
    genre: Optional[str] = None
    target_market: Optional[str] = None
    language: Optional[str] = None
    episode_count: Optional[int] = None
    stage: Optional[str] = None
    description: Optional[str] = None
    owner: Optional[str] = None
    priority: Optional[str] = None
    status: Optional[str] = None


class ShortDramaProjectOut(ShortDramaProjectBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ShortDramaProjectListResponse(BaseModel):
    # 列表接口返回 items + total，方便后续接分页器。
    items: list[ShortDramaProjectOut]
    total: int


class ProjectStats(BaseModel):
    # 项目总览统计：当前先兼容旧数据结构，后续会逐步改为真实 projectId 统计。
    content_plan_count: int
    episode_count: int
    script_count: int
    storyboard_count: int
    localization_count: int
    ad_material_count: int
    media_asset_count: int


class ProjectPipelineItem(BaseModel):
    key: str
    title: str
    status: str
    count: int
    description: str


class ProjectRecentItem(BaseModel):
    id: int
    title: str
    type: str
    status: str
    created_at: str


class ProjectOverview(BaseModel):
    project: ShortDramaProjectOut
    stats: ProjectStats
    pipeline: list[ProjectPipelineItem]
    recent_storyboards: list[ProjectRecentItem]
    recent_ad_materials: list[ProjectRecentItem]
    recent_media_assets: list[ProjectRecentItem]
